src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -49,7 +51,12 @@ def __init__(
         """
         if not _TORCH_INSTALLED:
             raise RuntimeWarning("This function requires PyTorch.")
```

Comment:
This feels a bit awkward, especially if it has to be done in each function using torch. Instead, one could try the imports in the `__init__.py` and import the right submodule based on that. E.g.
```python

try:
    import torch
    _TORCH_INSTALLED = True
except ImportError:
    _TORCH_INSTALLED = False
try:
    import jax
    _JAX_INSTALLED = True
except ImportError:
    _JAX_INSTALLED = False

if _TORCH_INSTALLED and FRAMEWORK == "torch":
    from frameworks.torch_differentiable import TorchTwiceDifferentiable as TwiceDifferentiable
    # etc. (or something like that)
elif _JAX_INSTALLED and FRAMEWORK == "jax":
     from frameworks.jax_differentiable import JaxTwiceDifferentiable as TwiceDifferentiable
     #...
```
This is however not very nice, because one must set the `FRAMEWORK` beforehand, maybe with an env variable or before importing the parent module, so that sucks.

Alternatively we could have submodules like:
```
influence/
|__torch.py
|__jax.py
|__tensorflow.py
|__ _common.py
```
and users would do
```python
import influence.torch as inf
inf.compute_influences(torch tensors and stuff)
```
Or 
```python
import influence.jax as inf
inf.compute_influences(jax data)
```
This will probably mean a lot of code duplication, though. And across different files. Using something like eagerpy might be tempting but those things just tend to die once the phd student defends or whatever (like eagerpy seems to have done).

Alternatively we could write some kind of `@dispatch` alternative, based on imports (or dispatch itself, based on the tensor types?) and keep all functions in the same file, so that at least they are bunched together...

Dunno, just spitballing here.. Ideas @AnesBenmerzoug @Xuzzo ?

Reply:
I went for the env variable. Let me know if it looks good enough

Reply:
Better solution proposed here #297